docs/topics/kotlin-daemon.md

```diff
@@ -0,0 +1,105 @@
+[//]: # (title: Kotlin daemon)
+
+The Kotlin daemon is a background process that build systems can use to improve Kotlin compilation performance.
+The Kotlin daemon speeds up build times by keeping the compiler and its environment ready to compile. This approach 
+avoids the overhead of starting a new Java Virtual Machine (JVM) instance and reinitializing the compiler for every 
+compilation, leading to reduced build times for incremental builds or frequent small changes.
+
+Some build systems, like [Gradle](https://docs.gradle.org/current/userguide/gradle_daemon.html) and [Maven](https://maven.apache.org/tools/mvnd.html),
+have their own daemons that help reduce startup costs. However, they miss the added benefit that the Kotlin daemon
+completely isolates the build system process from the compiler. This separation is useful in dynamic environments where
+system settings might change at runtime.
+
+Although there's no official user-facing interface for the daemon, it can be used by build systems or through the [build tools API](build-tools-api.md).
+
+## Kotlin daemon configuration
+
+There are ways to configure some settings for the Kotlin daemon for Gradle or Maven.
+
+### Memory management
+
+By default, the Kotlin daemon tries to inherit the heap size (`-Xmx`) of the launching JVM process.
+
+To configure specific memory limits, like `-Xmx` and `-XX:MaxMetaspaceSize`, use the following property:
+
+<tabs group="build-system">
+<tab title="Gradle" group-key="gradle">
+
+```kotlin
+kotlin.daemon.jvmargs=-Xmx1500m
+```
+
+For more information, see [`kotlin.daemon.jvmargs` property](gradle-compilation-and-caches.md#kotlin-daemon-jvmargs-property).
+
+</tab>
+<tab title="Maven" group-key="maven">
+
+```xml
+<kotlin.compiler.daemon.jvmArgs>-Xmx1500m</kotlin.compiler.daemon.jvmArgs>
+```
+
+</tab>
+</tabs>
+
+### Lifetime
+
+There are two common lifetime strategies for the Kotlin daemon:
+
+* **Short-lived daemon**: Used when the client (like a build system) is long-running. The daemon shuts down shortly after the
```

Comment:
I'm not sure about the naming "short-lived" and "long-lived".
Just as an idea, we may call them "attached" and "detached", where attached has its lifetime attached to the client process and detached is based on timeouts and last usage

Reply:
Works for me! Added.